### ISSUE TITLE
Module is not registered in "aliases" hooks

### ISSUE THREAD

**Issue by idnovate** [PrestaShop/PrestaShop#22527]
#### Describe the bug

Due to this modification:
https://github.com/PrestaShop/PrestaShop/commit/e59ebbddde684d7236b72aabbc2ce5bcb06d3364#diff-d5d273eeef7469e6cf4038eb2ee139ee12927a7709ea47f97d66e8b39da60178R587
https://github.com/PrestaShop/PrestaShop/blob/9a18f064ad50fd359485ec2b84313c9f393660f4/classes/Hook.php#L594

Since PS 1.7.7.0, if you try to register a module in a hook, it's not registered in their alias. Instead, a new hook is created.

The problem is that the second parameter is set to false (before PS 1.7.7.0 it was empty, true by default).

#### Expected behavior

Module should be registered in the module or their alias.

#### Steps to Reproduce

1. Install attached module
[dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/5726566/dummymodule.zip)

2. Check registered hooks
![image](https://user-images.githubusercontent.com/15104724/102801836-0f9d2d80-43b6-11eb-9dc1-6bac6d157a9a.png)
![image](https://user-images.githubusercontent.com/15104724/102801887-23e12a80-43b6-11eb-8e65-ad49e06548da.png)

Modules are registered in new hooks created, not in their alias

3. Same module in PS 1.7.6.9
![image](https://user-images.githubusercontent.com/15104724/102802865-9ef71080-43b7-11eb-9c74-1161d12c8841.png)
![image](https://user-images.githubusercontent.com/15104724/102802962-c6e67400-43b7-11eb-80c0-4a7ee361b931.png)

4. When hook is called, the module is NOT displayed if it's registered in hook `adminCustomers` or `adminOrder`

**Comment by PierreRambaud** [comment]
I manage to reproduce the issue with I had to change part of your module to make sure I understood what happened.
The problem comes from #12578 and not the merge request you target :sweat_smile: 

Here's a fix #22532

@PrestaShop/product-team I don't know if it's major or critical but it has to be merged for the 1.7.7.1 because some modules are maybe not working anymore if they use legacy names.

**Comment by PierreRambaud** [comment]
I update the module you uploaded with the fix to properly see the difference.
[dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/5726567/dummymodule.zip)

**Comment by marionf** [comment]
@PierreRambaud Is it a regression compared to 1.7.6.9 ?

**Comment by PierreRambaud** [comment]
> @PierreRambaud Is it a regression compared to 1.7.6.9 ?

Yes it is, and afaik there is no workaround to bypass the problem. It's maybe a critical issue!

**Comment by Progi1984** [comment]
Fixed by #22532

**Comment by imballo** [comment]
The citated change 
 $id_hook = Hook::getIdByName($hook_name, false); 
created to me a big problems.
I installed again a module that was using the taxManager alias (the hook is now called actionTaxManager) and it seemed that the module was skipped during the hook call.
After investigating I found that the citated above line inside Hook:registerHook not seeing the aliased one made registerHook create a new hook called taxManager.
Moreover execHookTaxManagerFactory in TaxManagerFactory calls the hook asking for the the modules connected to the alias "taxManager". The associative array cached by Hook:getAllHookIds inside Hook.php contains only one hook_id but the sql quering the database can return duplicated values (given the above behaviour) this translate in only one list of module called (the one connected to the aliased taxManager or the one connected to the taxManager hook recreated inside ps_hook).
